components/example-notebook-servers/jupyter-pytorch-gaudi/Dockerfile

```diff
@@ -0,0 +1,143 @@
+#
+# NOTE: Use the Makefiles to build this image correctly.
+#
+
+ARG BASE_IMG=<jupyter>
+FROM $BASE_IMG
+
+# Content below is extracted from scripts/Dockerfiles here:
+# https://github.com/HabanaAI/Setup_and_Install/tree/main/dockerfiles/base
+# https://github.com/HabanaAI/Setup_and_Install/tree/main/dockerfiles/pytorch
+
+# version details
+ARG ARTIFACTORY_URL=vault.habana.ai
+ARG VERSION=1.16.2
+ARG REVISION=2
+ARG PYTORCH_VERSION=2.2.2
+ARG OS_NUMBER=2204
+
+# runtime variables
+ENV DEBIAN_FRONTEND=noninteractive
+ENV GC_KERNEL_PATH=/usr/lib/habanalabs/libtpc_kernels.so
+ENV HABANA_LOGS=/var/log/habana_logs/
+ENV HABANA_SCAL_BIN_PATH=/opt/habanalabs/engines_fw
+ENV HABANA_PLUGINS_LIB_PATH=/opt/habanalabs/habana_plugins
+ENV TCMALLOC_LARGE_ALLOC_REPORT_THRESHOLD=7516192768
+ENV PYTHONPATH=/home/$NB_UID:/usr/lib/habanalabs/
```

Comment:
Is setting `PYTHONPATH` still necessary when we are sourcing `/etc/profile.d/habanalabs.sh` in bashrc/profile?

I only ask because having `PYTHONPATH` set can break things for end users in unexpected ways.

If it turns out to still be necessary, please move this it closer to the `RUN` where we install the files under `/usr/lib/habanalabs/`.

Reply:
PYTHONPATH is not set in the `habanalabs.sh` but I'll need to double check if it's really needed.

Reply:
PYTHONPATH was necessary for normal functionality.